### PR TITLE
fix: Fixed Typewriter component

### LINKED ISSUE #405: [BUG] <div> cannot appear as a descendant of <p>
### Is there any existing issue for this?

- [X] I have searched the existing issues

### Description

There is an error message in development mode. `TypeWriter Component` returns a `div`. This `div` is inside a `p` tag.

### Screenshots

console_error

### Any additional information?

_No response_

### What browser are you seeing the problem on?

Chrome

### PR DESCRIPTION
**Pull request by zvolcsey**
@rupali-codes @CBID2 Hi! I fixed one of the errors that appeared on the console. `TypeWriter component` returns a `div` and this `div` was inside a `p` tag.

## Fixes Issue

Closes #405 

## Changes proposed

- Removed parent p tag and added classes to the `Typewriter component`.

## Screenshots
console_error

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -26,14 +26,16 @@ export default function Home() {
             <span>👾</span>
           </div>
           <div className=" flex flex-col mt-6 justify-center items-start">
```
Comment (CBID2) [inline pages/index.tsx:28]:
```suggestion
          <article className=" flex flex-col mt-6 justify-center items-start">
```

File: pages/index.tsx [hunk 2]
```diff
@@ -26,14 +26,16 @@ export default function Home() {
...
+            <TypewriterComponent
+              options={{
+                strings: subCategoriesNames,
+                wrapperClassName: "text-xl text-violet-500",
+                cursorClassName: "text-xl text-violet-500",
+                autoStart: true,
+                loop: true,
+              }}
+            />
           </div>
```
Comment (CBID2) [inline pages/index.tsx:39]:
```suggestion
          </article>
```

Comment (CBID2) [inline pages/index.tsx:39]:
```suggestion
          </article>
```

File: pages/index.tsx [hunk 3]
```diff
@@ -26,14 +26,16 @@ export default function Home() {
             <span>👾</span>
           </div>
```
Comment (CBID2) [inline pages/index.tsx:27]:
```suggestion
          </article>
```

File: pages/index.tsx (hunk 2 above)
Comment (CBID2) [inline pages/index.tsx:39]:
```suggestion
          </article>
```

File: pages/index.tsx (hunk 1 above)
Comment (CBID2) [inline pages/index.tsx:28]:
```suggestion
          <article className=" flex flex-col mt-6 justify-center items-start">
```

File: pages/index.tsx (hunk 2 above)
Comment (CBID2) [inline pages/index.tsx:39]:
```suggestion
          </article>
```

File: pages/index.tsx (hunk 3 above)
Comment (CBID2) [inline pages/index.tsx:27]:
```suggestion
          </article>
```

Comment (CBID2) [inline pages/index.tsx:27]:
```suggestion
          </article>
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @zvolcsey! I made some suggestions on how to improve your code. It’s best to use `<div>`tags sparingly because they make your code less accessible.

**Comment by rupali-codes** [comment]
@CBID2 hey do you have any specific reason why are you replacing all <div> with <article>?

**Comment by CBID2** [comment]
> Hi @zvolcsey! I made some suggestions on how to improve your code. It’s best to use `<div>`tags sparingly because they make your code less accessible. Other than that, good job! :)

I explain why here @rupali-codes

**Comment by zvolcsey** [comment]
@rupali-codes @CBID2 Hi! May I ask what the decision is?
Should I change the other `div`-s? or
We keep the original `div`-s.
Only the parent `p` tag needs to be removed.

**Comment by CBID2** [comment]
> @rupali-codes @CBID2 Hi! May I ask what the decision is?
> 
> Should I change the other `div`-s? or
> 
> We keep the original `div`-s.
> 
> Only the parent `p` tag needs to be removed.

Good question @zvolcsey! I personally would use more semantic HTML to replace the `<div>`tags.

**Comment by rupali-codes** [comment]
> > @rupali-codes @CBID2 Hi! May I ask what the decision is?
> > Should I change the other `div`-s? or
> > We keep the original `div`-s.
> > Only the parent `p` tag needs to be removed.
> 
> Good question @zvolcsey! I personally would use more semantic HTML to replace the `<div>`tags.

Umm..But then we'll have to replace it from everywhere, @Eugene44-hub do you have any takes on this?

**Comment by Eugene44-hub** [comment]
> > > @rupali-codes @CBID2 Hi! May I ask what the decision is?
> > > Should I change the other `div`-s? or
> > > We keep the original `div`-s.
> > > Only the parent `p` tag needs to be removed.
> > 
> > 
> > Good question @zvolcsey! I personally would use more semantic HTML to replace the `<div>`tags.
> 
> Umm..But then we'll have to replace it from everywhere, @Eugene44-hub do you have any takes on this?

So i suggest the article tag should be used like where we have much text. Like the cards that will be ok.

**Comment by zvolcsey** [comment]
@rupali-codes @CBID2 @Eugene44-hub Thanks for the feedback! Then I will replace the `article` wtih the original `div`. The issue is soldved without changing the tags and we can improve the structure and the tags later.

**Comment by zvolcsey** [comment]
@rupali-codes @CBID2 @Eugene44-hub Hi! In my latest commits, I made the followings improvements:

- Merge the latest commits.
- Replaced the changed `article` tag with the original `div`.

The currently changes:

- Format with Prettier.
- Removed the parent `p` tag and added the styles for the `Typewriter-component`.

I think we can improve the tags in an other issue, if we will find a better solution. The issue is currently solved. The error message disappeared in development mode, because the `p` tag does not have a `div` tag.

**Comment by CBID2** [comment]
> @rupali-codes @CBID2 @Eugene44-hub Hi! In my latest commits, I made the followings improvements:
> 
> 
> 
> - Merge the latest commits.
> 
> - Replaced the changed `article` tag with the original `div`.
> 
> 
> 
> The currently changes:
> 
> 
> 
> - Format with Prettier.
> 
> - Removed the parent `p` tag and added the styles for the `Typewriter-component`.
> 
> 
> 
> I think we can improve the tags in an other issue, if we will find a better solution. The issue is currently solved. The error message disappeared in development mode, because the `p` tag does not have a `div` tag.

Ok that's a good idea

**Review by rupali-codes — COMMENTED** [review]
looks good to me, could you please resolve this conversations?

**Comment by zvolcsey** [comment]
@rupali-codes Hi! In my latest commit I resolved the merge conflict.

The problem was the `p` tag before the `TypewriterComponent`. 
The `TypewriterComponent` returns a `div`.
In development mode was an error, because the `p` tag had a `div` tag.

**Comment by rupali-codes** [comment]
> @rupali-codes Hi! In my latest commit I resolved the merge conflict.
> 
> The problem was the `p` tag before the `TypewriterComponent`. The `TypewriterComponent` returns a `div`. In development mode was an error, because the `p` tag had a `div` tag.

Thanks

**Pull request merged**